rcldotnet/Clock.cs

```diff
@@ -0,0 +1,261 @@
+            };
+        }
+    }
+
+    public class CallbackAlreadyRegisteredException : Exception
+    {
+        public CallbackAlreadyRegisteredException(string message) : base(message)
+        {
+        }
+    }
+
+    internal static class ClockDelegates
+    {
+        private static readonly DllLoadUtils _dllLoadUtils;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        internal delegate RCLRet NativeRCLClockFunctionType(SafeClockHandle clockHandle);
+
+        internal static NativeRCLClockFunctionType native_rcl_enable_ros_time_override = null;
+
+        internal static NativeRCLClockFunctionType native_rcl_disable_ros_time_override = null;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        internal delegate RCLRet NativeRCLClockGetNowType(SafeClockHandle clockHandle, out TimePoint time);
+
+        internal static NativeRCLClockGetNowType native_rcl_clock_get_now = null;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        internal delegate RCLRet NativeRCLSetRosTimeOverrideType(SafeClockHandle clockHandle, long timePointValue);
+
+        internal static NativeRCLSetRosTimeOverrideType native_rcl_set_ros_time_override = null;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        internal delegate RCLRet NativeRCLAddJumpCallbackType(SafeClockHandle clockHandle, JumpThreshold threshold, JumpCallbackInternal callback);
+
+        internal static NativeRCLAddJumpCallbackType native_rcl_clock_add_jump_callback = null;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        internal delegate RCLRet NativeRCLRemoveJumpCallbackType(SafeClockHandle clockHandle, JumpCallbackInternal callback);
+
+        internal static NativeRCLRemoveJumpCallbackType native_rcl_clock_remove_jump_callback = null;
+
+        static ClockDelegates()
+        {
+            _dllLoadUtils = DllLoadUtilsFactory.GetDllLoadUtils();
+            IntPtr nativeLibrary = _dllLoadUtils.LoadLibrary("rcldotnet");
+
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_enable_ros_time_override), out native_rcl_enable_ros_time_override);
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_disable_ros_time_override), out native_rcl_disable_ros_time_override);
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_clock_get_now), out native_rcl_clock_get_now);
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_set_ros_time_override), out native_rcl_set_ros_time_override);
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_clock_add_jump_callback), out native_rcl_clock_add_jump_callback);
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_clock_remove_jump_callback), out native_rcl_clock_remove_jump_callback);
+        }
+    }
+
+    public sealed class Clock
```

Comment:
It's an interesting proposition, it does have the potential to eliminate confusion. I am finding I'm fully qualifying my usages of `rcldotnet` message types for this reason.

If we did make this change though, it would be a deviation from both `rclcpp` and `rclpy`.